File: linked-list/reverse-linked-list.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode() : val(0), next(nullptr) {}
 *     ListNode(int x) : val(x), next(nullptr) {}
 *     ListNode(int x, ListNode *next) : val(x), next(next) {}
 * };
 */
class Solution {
public:
    ListNode* reverseList(ListNode* head) {
        if(head==nullptr){
            return NULL;
        }
        ListNode* node = new ListNode(head->val);
        ListNode* cur = head->next;
        while(cur!=nullptr){
            ListNode* newNode = new ListNode(cur->val);
            newNode->next = node;
            node = newNode;
            cur=cur->next;
        }
        return node;
    }
};

File: linked-list/reverse-linked-list.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */
class Solution {
    public ListNode reverseList(ListNode head) {
        if (head == null){
            return null; 
        }
        ListNode cur = head.next; // Use to traverse head
        ListNode reverse = new ListNode(head.val); // Use to store reversed LinkedList
        while(cur != null){
            ListNode prev = new ListNode(cur.val); 
            prev.next = reverse; 
            reverse = prev; 
            cur = cur.next; 
        }
        return reverse; 
    }
}